mlir/lib/Dialect/SPIRV/Transforms/SPIRVConversion.cpp

```diff
@@ -34,6 +44,38 @@ using namespace mlir;
 // Utility functions
 //===----------------------------------------------------------------------===//
 
+static int getComputeVectorSize(int64_t size) {
+  for (int i : {4, 3, 2}) {
```

Comment:
OpenCL SPIR-V supports 2, 3, 4, 8, or 16 as vector sizes and some SPIR-V environments (ahem, Intel) supports even more. It would be nice to make it configurable.

Reply:
Nevermind, didn't noticed first comment